File: backend/core/src/modules/world-settings/world-settings.controller.ts
import { Controller, UseGuards } from '@nestjs/common';
import { ProjectScopedCrudController } from '../../common/access/project-scoped-crud.controller';
import { WorldSettingsService } from './world-settings.service';
import { WorldSetting } from '../../entities';
import { JwtAuthGuard } from '../auth/jwt-auth.guard';

@UseGuards(JwtAuthGuard)
@Controller('world-settings')
export class WorldSettingsController extends ProjectScopedCrudController<WorldSetting> {
  constructor(service: WorldSettingsService) {
    super(service);
  }
}

File: backend/core/src/main/java/store/pjcloud/cowrite/core/controller/WorldSettingsController.java
package store.pjcloud.cowrite.core.controller;

import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import store.pjcloud.cowrite.core.common.ProjectScopedCrudController;
import store.pjcloud.cowrite.core.entity.WorldSetting;
import store.pjcloud.cowrite.core.service.WorldSettingsService;

@RestController
@RequestMapping("/world-settings")
public class WorldSettingsController extends ProjectScopedCrudController<WorldSetting> {
    public WorldSettingsController(WorldSettingsService service) {
        super(service);
    }
}
